src/NServiceBus.RavenDB/Timeouts/QueryTimeouts.cs

```diff
@@ -28,6 +29,11 @@ public QueryTimeouts(IDocumentStore store)
 
         public async Task<TimeoutsChunk> GetNextChunk(DateTime startSlice)
         {
+            if(abort)
+            {
+                return new TimeoutsChunk(new TimeoutsChunk.Timeout[ 0 ], DateTime.UtcNow.AddHours(1));
```

Comment:
What about moving this to the end of the method? We'll do a little extra work if `Shutdown` is called but it will effectively get rid of the window where a bunch of timeouts might get picked up. We could also have it in both places, but that seems gross.